src/macros.rs

```diff
@@ -1,3 +1,40 @@
+/// Create an [`IndexMap`][crate::IndexMap] from a list of key-value pairs  
+/// and a `BuildHasherDefault`-wrapped custom hasher.  
+///
+/// ## Example
+///
+/// ```
+/// use indexmap::indexmap_with_default;
+/// use fnv::FnvHasher;
+///
+/// let map = indexmap_with_default!{
+///     FnvHasher;
+///     "a" => 1,
+///     "b" => 2,
+/// };
+/// assert_eq!(map["a"], 1);
+/// assert_eq!(map["b"], 2);
+/// assert_eq!(map.get("c"), None);
+///
+/// // "a" is the first key
+/// assert_eq!(map.keys().next(), Some(&"a"));
+/// ```
+#[macro_export]
+macro_rules! indexmap_with_default {
+    ($H:ty; $($key:expr => $value:expr,)+) => { $crate::indexmap_with_default!($H; $($key => $value),+) };
+    ($H:ty; $($key:expr => $value:expr),*) => {{
+        let builder = ::core::hash::BuildHasherDefault::<$H>::new();
```

Comment:
Oh, I didn't realize `new()` was *just* stabilized in Rust 1.85. (https://github.com/rust-lang/rust/issues/123197)

Better go back to `default()` then.
```suggestion
        let builder = ::core::hash::BuildHasherDefault::<$H>::default();
```